test/handlers/UpgradeRegressionManager.handler.sol

```diff
@@ -0,0 +1,264 @@
+    } catch {}
+  }
+
+  /// @notice Get a random rollback ID
+  /// @param _randomIndex Used to randomly select a proposal
+  /// @return The rollback ID of a randomly selected proposal, or 0 if none exist
+  function getRandomRollbackId(uint256 _randomIndex) public view returns (uint256) {
+    uint256 _index = bound(_randomIndex, 0, rollbackIds.length - 1);
+    return rollbackIds[_index];
+  }
+
+  /// @notice Get a random queueable rollback ID
+  /// @param _randomIndex Used to randomly select a queueable proposal
+  /// @return The rollback ID of a randomly selected queueable proposal, or 0 if none exist
+  function getRandomQueueableRollbackId(uint256 _randomIndex) public view returns (uint256) {
+    uint256[] memory _matchingIds = new uint256[](rollbackIds.length);
+    uint256 _matchCount = 0;
+
+    for (uint256 i = 0; i < rollbackIds.length; i++) {
+      if (proposals[rollbackIds[i]].state == ProposalState.PROPOSED) {
+        _matchingIds[_matchCount] = rollbackIds[i];
+        _matchCount++;
+      }
+    }
+
+    if (_matchCount == 0) {
+      return 0;
+    }
+
+    uint256 _selectedIndex = _randomIndex % _matchCount;
+    return _matchingIds[_selectedIndex];
+  }
+
+  /// @notice Get a random executable rollback ID
+  /// @param _randomIndex Used to randomly select an executable proposal
+  /// @return The rollback ID of a randomly selected executable proposal, or 0 if none exist
+  function getRandomExecutableRollbackId(uint256 _randomIndex) public view returns (uint256) {
+    uint256[] memory _matchingIds = new uint256[](rollbackIds.length);
+    uint256 _matchCount = 0;
+
+    for (uint256 i = 0; i < rollbackIds.length; i++) {
+      if (proposals[rollbackIds[i]].state == ProposalState.QUEUED) {
+        _matchingIds[_matchCount] = rollbackIds[i];
+        _matchCount++;
+      }
+    }
+
+    if (_matchCount == 0) {
+      return 0;
+    }
+
+    uint256 _selectedIndex = _randomIndex % _matchCount;
+    return _matchingIds[_selectedIndex];
+  }
+
+  /// @notice Warp time to expire a proposed rollback
+  /// @param _randomIndex Used to randomly select a proposed proposal
+  /// @return The rollback ID of the expired rollback, or 0 if none found
+  function expireRollback(uint256 _randomIndex) public returns (uint256) {
+    uint256 _rollbackId = getRandomQueueableRollbackId(_randomIndex);
+    if (_rollbackId == 0) {
+      return 0;
+    }
+
+    // Get the expiry time for this rollback
+    uint256 _expiresAt = urm.rollbackQueueExpiresAt(_rollbackId);
+    if (_expiresAt == 0) {
+      return 0; // Not in queueing state
+    }
+
+    // Warp time past the expiry window
+    vm.warp(_expiresAt + 1);
+
+    return _rollbackId;
+  }
+
+  /// @notice Warp time to make a queued rollback executable
+  /// @param _randomIndex Used to randomly select a queued proposal
+  /// @return The rollback ID of the executable rollback, or 0 if none found
+  function makeRollbackExecutable(uint256 _randomIndex) public returns (uint256) {
```

Comment:
Per my other comment, we shouldn't have a handler that specifically makes a certain rollback executable. Instead we should just have a handler that advances time, and thus proposals become executable.